core/primitives/src/shard_layout.rs

```diff
@@ -34,9 +34,11 @@ pub struct ShardLayoutV1 {
     fixed_shards: Vec<AccountId>,
     /// The rest are divided by boundary_accounts to ranges, each range is mapped to a shard
     boundary_accounts: Vec<AccountId>,
-    /// Parent shards for the shards, useful for constructing states for the shards.
+    /// Maps shards from the last shard layout to shards that it splits to in this shard layout,
+    /// Instead of using map, we just use a vec here because shard_id ranges from 0 to num_shards-1
+    /// Useful for constructing states for the shards.
     /// None for the genesis shard layout
-    parent_shards: Option<Vec<ShardId>>,
+    shards_split_map: Option<Vec<Vec<ShardId>>>,
```

Comment:
Consider having a separate type since `Vec<Vec<>>` looks suspicious, and I had to read the comment to understand the data layout (the meaning of the indices)